src/unix/socket.c

```diff
@@ -571,21 +633,33 @@ static sysreturn unixsock_accept4(struct sock *sock, struct sockaddr *addr,
 sysreturn unixsock_sendto(struct sock *sock, void *buf, u64 len, int flags,
         struct sockaddr *dest_addr, socklen_t addrlen)
 {
-    /* Non-connected sockets are not supported, so destination address is
-     * ignored. */
-    return apply(sock->f.write, buf, len, 0, current, false,
-            syscall_io_complete);
+    unixsock s = (unixsock) sock;
+    if (dest_addr || addrlen) {
+        if (sock->type == SOCK_STREAM) {
+            if (s->peer)
+                return -EISCONN;
+            else
+                return -ENOTCONN;
+        }
+        if (!(dest_addr && addrlen))
+            return -EFAULT;
+        if (addrlen < sizeof(struct sockaddr_un))
+            return -EINVAL;
+        if (dest_addr && !validate_user_memory(dest_addr, addrlen, true))
```

Comment:
change true to false here if you end up removing the null terminator write

Reply:
I actually removed it completely because it is already correctly checked in the main sendto function.